x/rewards/keeper/grpc_query.go

```diff
@@ -0,0 +1,60 @@
+package keeper
+
+import (
+	"context"
+	"github.com/archway-network/archway/x/rewards/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var _ types.QueryServer = &QueryServer{}
+
+// QueryServer implements the module gRPC query service.
+type QueryServer struct {
+	keeper Keeper
+}
+
+// NewQueryServer creates a new gRPC query server.
+func NewQueryServer(keeper Keeper) *QueryServer {
+	return &QueryServer{
+		keeper: keeper,
+	}
+}
+
+// Params implements the types.QueryServer interface.
+func (s *QueryServer) Params(c context.Context, request *types.QueryParamsRequest) (*types.QueryParamsResponse, error) {
+	if request == nil {
```

Comment:
I think we can avoid checking for this as the gRPC stub should handle this seamlessly

Reply:
This is done in the Cosmos SDK so I did the same 🤷‍♂️